x-pack/test/security_solution_api_integration/test_suites/detections_response/rule_creation/create_rules.ts

```diff
@@ -30,43 +30,56 @@ import {
   getSimpleMlRule,
   getSimpleMlRuleOutput,
   waitForRuleSuccess,
-  getRuleForSignalTesting,
-  getRuleForSignalTestingWithTimestampOverride,
+  getRuleForAlertTesting,
+  getRuleForAlertTestingWithTimestampOverride,
   waitForAlertToComplete,
-  waitForSignalsToBePresent,
-  getThresholdRuleForSignalTesting,
+  waitForAlertsToBePresent,
+  getThresholdRuleForAlertTesting,
   waitForRulePartialFailure,
   createRule,
   deleteAllAlerts,
-} from '../../utils';
-import { createUserAndRole, deleteUserAndRole } from '../../../common/services/security_solution';
-import {
+  removeUUIDFromActions,
   getActionsWithFrequencies,
   getActionsWithoutFrequencies,
   getSomeActionsWithFrequencies,
-} from '../../utils/get_rule_actions';
-import { removeUUIDFromActions } from '../../utils/remove_uuid_from_actions';
+  updateUsername,
+} from '../utils';
+import {
+  createUserAndRole,
+  deleteUserAndRole,
+} from '../../../../common/services/security_solution';
 
-// eslint-disable-next-line import/no-default-export
 export default ({ getService }: FtrProviderContext) => {
   const esArchiver = getService('esArchiver');
   const supertest = getService('supertest');
   const supertestWithoutAuth = getService('supertestWithoutAuth');
   const log = getService('log');
   const es = getService('es');
+  const config = getService('config');
+  const { ELASTICSEARCH_USERNAME } = config.get('kbnTestServer.env');
+  const isServerless = config.get('serverless');
 
-  describe('create_rules', () => {
+  describe('@serverless @ess create_rules', () => {
     describe('creating rules', () => {
       before(async () => {
-        await esArchiver.load('x-pack/test/functional/es_archives/auditbeat/hosts');
+        // TODO ask about best approach here
+        if (isServerless)
+          await esArchiver.load(
+            'x-pack/test/security_solution_api_integration/es_archive/serverless/host_serverless'
+          );
+        else await esArchiver.load('x-pack/test/functional/es_archives/auditbeat/hosts');
```

Comment:
Just responding to the `TODO:` here. I'm thinking maybe some utility class that contains some options in the constructor to determine what archived data to load / unload. So when the class is instantiated with the `isServerless` it will load the serverless hots data, and maybe include some of the other decorators you added in this PR in the constructor too? Could be a neat little testing utility to have available and would lessen the likelihood of repeated branching logic within individual tests. I can see that branching logic growing (and becoming unwieldy) as we begin to separate code paths between serverless and on-prem.

Reply:
Thanks @dhurley14 for your input 🙏🏻 was thinking of a utility function but the class makes more sense, is this how you imagined it? https://github.com/elastic/kibana/pull/166755/commits/3320fd193a2ce19d61a3a37952e077dbc0d524d8

Reply:
That looks great! Better than I imagined haha.

Reply:
Thanks for the inspiration!! 😉